### PR TITLE
test(flow): increase code coverage with tests

### LINKED ISSUE #2068: increase code coverage with tests
## Summary
Track the missing compatibility and behavior tests identified while comparing `falkordb/falkordb:edge` with `ghcr.io/falkordb/falkordb:edge-rs`.

Connected PR: #2067

## Missing tests covered by the connected PR
- Raw `GRAPH.QUERY` response contracts: plain string headers, scalar/list/map/entity/vector raw shapes, and no `Graph version` footer stat.
- Mutation metadata contracts: `CREATE INDEX` stats, `GRAPH.CONSTRAINT CREATE` returning `PENDING`, and unsupported `GRAPH.CONSTRAINT LIST` arity behavior.
- Redis command metadata contracts for graph commands, including `denyoom` and `allow_busy` flags.
- Raw execution-plan roots for `GRAPH.EXPLAIN` and `GRAPH.PROFILE`.
- `dbms.functions()` catalog entries that were missing or different across implementations.
- Temporal contracts for `duration('P1M')` formatting and month-end date/localdatetime arithmetic.
- Error/boundary contracts for `abs(INT64_MIN)`, unsupported `=~`, `ASYNC_DELETE` runtime set failure, whitespace/semicolon-only queries, wrong-type `GRAPH.DELETE`, duplicate `GRAPH.COPY`, and procedure lookup/yield errors.
- `CALL db.indexes()` raw catalog column order and default fulltext `language`/`stopwords` fields.
- `GRAPH.SLOWLOG` timestamp integer-format contract.

## Notes
Performance regressions found in the comparison are intentionally not tracked as flow-test assertions here because wall-clock thresholds would be flaky in CI. They should be handled by benchmark infrastructure separately.

### PR DESCRIPTION
**Pull request by gkorland**
## Summary
Add flow-test coverage for compatibility gaps found while comparing `falkordb/falkordb:edge` with `ghcr.io/falkordb/falkordb:edge-rs`.

Fix: #2068

## Changes
- Added the missing compatibility scenarios to the relevant existing flow suites instead of a standalone compatibility file:
  - `test_results.py` for raw response/header/entity/vector/footer shape.
  - `test_index_create.py`, `test_constraint.py`, and `test_procedures.py` for index stats, constraint status/arity, and catalog procedure output.
  - `test_commands_registration.py` and `test_profile.py` for command metadata and raw plan roots.
  - `test_temporal.py`, `test_function_calls.py`, `test_config.py`, `test_empty_query.py`, `test_keyspace_accesses.py`, `test_graph_copy.py`, and `test_slowlog.py` for the remaining behavior/error contracts.
- Verified the added scenarios were not already covered as exact contracts by existing flow tests before adding them; existing tests mostly covered semantic success/error paths but not raw wire shape, metadata flags, footer stats, or exact compatibility-sensitive text.
- Kept performance drawbacks out of flow-test thresholds to avoid flaky CI timing assertions; benchmark gating should be handled separately in benchmark infrastructure.

## Testing
- `python3 -m py_compile tests/flow/test_results.py tests/flow/test_index_create.py tests/flow/test_constraint.py tests/flow/test_commands_registration.py tests/flow/test_profile.py tests/flow/test_procedures.py tests/flow/test_temporal.py tests/flow/test_function_calls.py tests/flow/test_config.py tests/flow/test_empty_query.py tests/flow/test_keyspace_accesses.py tests/flow/test_graph_copy.py tests/flow/test_slowlog.py`
- `MODULE=$PWD/bin/linux-x64-release/falkordb.so TEST="test_results test_index_create test_constraint test_commands_registration test_profile test_procedures test_function_calls test_config test_empty_query test_keyspace_accesses test_graph_copy test_slowlog" AOF=0 TCK=0 UPGRADE=0 ./tests/flow/tests.sh`
- `MODULE=$PWD/bin/linux-x64-release/falkordb.so TEST="test_temporal:testTemporalDuration.test_duration_from_string test_temporal:testTemporalDuration.test_month_end_duration_arithmetic" AOF=0 TCK=0 UPGRADE=0 ./tests/flow/tests.sh`
- `git diff --check`

## Memory / Performance Impact
N/A; test-only change.

## Related Issues
References #2068

## Summary by CodeRabbit

* **Tests**
  * Added validation for command metadata flags and config-setting failures
  * Expanded constraint, index-creation, index-catalog, and keyspace-access checks
  * Strengthened parsing, procedure, and function-call error assertions (including unsupported regex)
  * Added raw response / profiling contract and result-formatting validations
  * Extended temporal month-end duration arithmetic and overflow checks
  * Added whitespace/semicolon empty-query tests and slowlog entry format validation

[Review Change Stack](https://app.coderabbit.ai/change-stack/FalkorDB/FalkorDB/pull/2067?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: tests/flow/test_commands_registration.py [hunk 1]
```diff
@@ -39,3 +46,12 @@ def test_deny_script(self):
...
 
+    def test_command_metadata_flags(self):
+        self.env.assertIn("denyoom", self._command_flags("graph.query"))
+        self.env.assertIn("denyoom", self._command_flags("graph.profile"))
+        self.env.assertIn("denyoom", self._command_flags("graph.constraint"))
+        self.env.assertIn("denyoom", self._command_flags("graph.copy"))
+
+        self.env.assertIn("allow_busy", self._command_flags("graph.config"))
+        self.env.assertIn("allow_busy", self._command_flags("graph.slowlog"))
+        self.env.assertNotIn("allow_busy", self._command_flags("graph.query"))
```
Comment (swilly22) [inline tests/flow/test_commands_registration.py]: Too specific, please remove.

File: tests/flow/test_index_create.py [hunk 2]
```diff
@@ -819,3 +819,18 @@ def test15_index_progress_report(self):
...
+    def test16_index_creation_raw_stats(self):
+        graph_name = "index_create_stats_contract"
+        redis_con = self.env.getConnection()
+        redis_con.delete(graph_name)
+
+        result = redis_con.execute_command(
+            "GRAPH.QUERY",
+            graph_name,
+            "CREATE INDEX FOR (n:N) ON (n.a)",
+        )
```
Comment (swilly22) [inline tests/flow/test_index_create.py]: Avoid issuing queries via a direct redis connection, this is why we have the FalkorDB Graph object in addition to index creation dedicated functions.

File: tests/flow/test_index_create.py [hunk 3]
```diff
@@ -819,3 +819,18 @@ def test15_index_progress_report(self):
...
+        stats = result[-1]
```
Comment (swilly22) [inline tests/flow/test_index_create.py]: when using the graph object query method, you'll get a result set statistics object, which should be used to query and assert statistics.

File: tests/flow/test_procedures.py [hunk 4]
```diff
@@ -423,6 +423,21 @@ def test12_list_functions(self):
...
+            "endNode",
+            "property",
+            "randomuuid",
+            "startNode",
+            "string.matchRegEx",
+            "typeof",
+            "xor",
+        ]:
+            self.env.assertIn(name, names)
+
```
Comment (swilly22) [inline tests/flow/test_procedures.py]: are we trying to cover all built-in functions? if yes, there are plenty more.
I think this is unnecessary and should be removed.

File: tests/flow/test_procedures.py [hunk 5]
```diff
@@ -474,3 +489,34 @@ def test12_list_functions(self):
         self.env.assertEquals(aggregation, False)
         self.env.assertEquals(variable_len, True)
         self.env.assertEquals(udf, False)
+
+    def test13_index_catalog_raw_response(self):
```
Comment (swilly22) [inline tests/flow/test_procedures.py]: Should this test function be in `test_index_create.py` ?

File: tests/flow/test_results.py [hunk 6]
```diff
@@ -1,3 +1,5 @@
+import re
```
Comment (swilly22) [inline tests/flow/test_results.py]: I don't see the value in the changes made to this file, please revert all of them.

File: tests/flow/test_slowlog.py [hunk 7]
```diff
@@ -1,4 +1,5 @@
 import asyncio
+import re
```
Comment (swilly22) [inline tests/flow/test_slowlog.py]: changes in this file are unnecessary, please revert.

File: tests/flow/test_temporal.py [hunk 8]
```diff
@@ -446,6 +447,24 @@ def test_duration_from_string(self):
             actual = result.result_set[0][0]
             self.env.assertEquals(actual, expected)
 
+        raw = self.env.getConnection().execute_command(
+            "GRAPH.QUERY",
+            GRAPH_ID,
+            "RETURN duration('P1M') AS d, toString(duration('P1M')) AS s",
+        )
+        self.env.assertEquals(raw[1], [["P1M", "P1M"]])
```
Comment (swilly22) [inline tests/flow/test_temporal.py]: use the graph.query function to issue queries, do not send queries directly via `execute_command`

File: tests/flow/test_temporal.py [hunk 9]
```diff
@@ -446,6 +447,24 @@ def test_duration_from_string(self):
...
+
+    def test_month_end_duration_arithmetic(self):
+        raw = self.env.getConnection().execute_command(
```
Comment (swilly22) [inline tests/flow/test_temporal.py]: use the graph.query function to issue queries, do not send queries directly via execute_command

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: tests/flow/test_commands_registration.py (hunk 1 above)
Comment (gkorland) [reply tests/flow/test_commands_registration.py]: @swilly22 but that is here to verify this behaviour is kept and not changed accidentally by a bug

File: tests/flow/test_procedures.py (hunk 4 above)
Comment (gkorland) [reply tests/flow/test_procedures.py]: we need to cover it all and make sure this behaviour is consistent

File: tests/flow/test_empty_query.py [hunk 10]
```diff
@@ -12,6 +12,14 @@ def test01_empty_query(self):
         except ResponseError as e:
             self.env.assertIn("Error: empty query.", str(e))
 
+    def test02_whitespace_and_semicolon_queries(self):
+        for query in [" ", ";"]:
+            try:
+                self.graph.query(query)
+                self.env.assertTrue(False)
+            except ResponseError as e:
+                self.env.assertIn("Error: could not parse query", str(e))
```
[the remark this answers is not in the document]
Comment (gkorland) [reply tests/flow/test_empty_query.py]: @copilot are you sure this assert as is will fail? it seems like the CI is passing without issues.

File: tests/flow/test_index_create.py (hunk 3 above)
Comment (gkorland) [reply tests/flow/test_index_create.py]: @swilly22  I'm aware but this test is to verify the raw result in the protocol while using the client object might work even when the server protocol is not strictly kept

File: tests/flow/test_index_create.py (hunk 2 above)
Comment (gkorland) [reply tests/flow/test_index_create.py]: Addressed in edf0dce5e: `test16_index_creation_stats` now uses the Graph/index helper (`create_node_range_index`) instead of issuing `GRAPH.QUERY` through a raw Redis connection.

File: tests/flow/test_procedures.py (hunk 5 above)
Comment (gkorland) [reply tests/flow/test_procedures.py]: Addressed in edf0dce5e: moved this coverage out of `test_procedures.py` and into `test_index_create.py` as `test17_index_catalog_response`, using `graph.query` for the catalog read.

File: tests/flow/test_results.py (hunk 6 above)
Comment (gkorland) [reply tests/flow/test_results.py]: Addressed in edf0dce5e: reverted the changes to `tests/flow/test_results.py`.

File: tests/flow/test_slowlog.py (hunk 7 above)
Comment (gkorland) [reply tests/flow/test_slowlog.py]: Addressed in edf0dce5e: reverted the changes to `tests/flow/test_slowlog.py`.

File: tests/flow/test_temporal.py (hunk 8 above)
Comment (gkorland) [reply tests/flow/test_temporal.py]: Addressed in edf0dce5e: replaced the raw `execute_command("GRAPH.QUERY", ...)` call with `self.graph.query(...)` in the duration string coverage.

File: tests/flow/test_temporal.py (hunk 9 above)
Comment (gkorland) [reply tests/flow/test_temporal.py]: Addressed in edf0dce5e: replaced the raw `execute_command("GRAPH.QUERY", ...)` call with `self.graph.query(...)` in the month-end duration arithmetic coverage.

File: tests/flow/test_index_create.py (hunk 3 above)
Comment (swilly22) [reply tests/flow/test_index_create.py]: Then this is defiantly not the place to test that.
if we want to verify the object's structure we need a dedicated test file for that.

File: tests/flow/test_procedures.py (hunk 4 above)
Comment (swilly22) [reply tests/flow/test_procedures.py]: Well this is not the entire list.
I think its way too much, also the list can change when UDFs are registered.

**Review by swilly22 — APPROVED** [review]
